syntax = "proto3";
option cc_enable_arenas = true;
package NetProto;

enum TryEnum
{
	a = 0;
	b = 1;
}

message TryItem
{
	int32 id = 1;
	string name = 2;
}

message TryMsg
{
	enum TryNestEnum
	{
		a = 0;
		b = 1;
	}	
	
	int32 int_val = 1;
	string str_val = 2;
	TryItem msg_val = 3;
	TryEnum enum_val = 4;
	TryNestEnum nest_enum_val = 5;
	repeated TryItem msg_array = 6;
	repeated TryNestEnum nest_enum_array = 7;
}